src/client/Client.cc

```diff
@@ -8779,15 +8763,51 @@ int Client::statfs(const char *path, struct statvfs *stbuf)
   const int CEPH_BLOCK_SHIFT = 22;
   stbuf->f_frsize = 1 << CEPH_BLOCK_SHIFT;
   stbuf->f_bsize = 1 << CEPH_BLOCK_SHIFT;
-  stbuf->f_blocks = stats.kb >> (CEPH_BLOCK_SHIFT - 10);
-  stbuf->f_bfree = stats.kb_avail >> (CEPH_BLOCK_SHIFT - 10);
-  stbuf->f_bavail = stats.kb_avail >> (CEPH_BLOCK_SHIFT - 10);
   stbuf->f_files = stats.num_objects;
   stbuf->f_ffree = -1;
   stbuf->f_favail = -1;
   stbuf->f_fsid = -1;       // ??
   stbuf->f_flag = 0;        // ??
-  stbuf->f_namemax = NAME_MAX;
+
+  C_SaferCond cond;
+  objecter->get_fs_stats(stats, &cond);
+
+  client_lock.Unlock();
+  int rval = cond.wait();
+  client_lock.Lock();
+
+  if (cct->_conf->client_quota && root_ancestor
+      && root_ancestor->quota.max_bytes) {
```

Comment:
I think root_ancestor->quota is filesystem root's quota. It's possible that the mounted sub-directory or its ancestor directories have quota